src/imageruler/imageruler.py

```diff
@@ -0,0 +1,620 @@
+"""Imageruler for measuring minimum lengthscales in binary images."""
+
+import dataclasses
+import enum
+import functools
+from typing import Any, Callable, Tuple
+
+import cv2
+import numpy as onp
+
+NDArray = onp.ndarray[Any, Any]
+
+
+FEASIBILITY_GAP_ALLOWANCE = 10
```

Comment:
DEFAULT_FEASIBILITY_GAP_ALLOWANCE
might be clearer

Reply:
Also, is `10` a reasonable default value for this? Maybe you can add a docstring here (or down in the function) on how one might want to choose this parameter or when one might want to change it?

Reply:
The docstring can be improved, but just answering here for reference

>  In checking whether a `x` is feasible with a brush of size `n`, we also check for feasibility with larger brushes, since e.g. some features realizable with a brush `n + k` may not be realizable with the brush of size `n`. The `feasibility_gap_allowance is the maximum value of `k` used.

If the true feature size of your design is very large, _and you are not ignoring any violations_, then larger k may be required.

The connection to the ignore scheme is as follows: in general, if a design is feasible with `n + k` but not `n`, the violations are on the borders of the feature. And, if we are ignoring those anyway, then the design will be deemed feasible with `n`, even if this is not strictly true.

As a follow-up, one could add to the `advanced.ipynb` notebook to demonstrate the effect of this feature.